### DISCUSSION TITLE
Works intermittently on chrome on Linux Mint

### DISCUSSION THREAD

**Discussion by BjornFJohansson** [coddingtonbear/obsidian-web#170] [Bug Report]
### Describe the bug

All the settings are correct afaik, see screenshots below. Stops working after browser restart. First screenshots to prove that the settings are correct to the best of my understanding, symptoms of the problem at the end as well as log in json format. When it works, this is a great tool and time saver. Thanks for your effort. /bjorn

![Screenshot from 2024-02-27 06-14-25](https://github.com/coddingtonbear/obsidian-web/assets/9726401/7d0d56c2-36af-4c95-a566-9f9fb806617f)

![Screenshot from 2024-02-27 06-17-55](https://github.com/coddingtonbear/obsidian-web/assets/9726401/fa4736e6-834d-47b9-abd7-9997abf21631)

![Screenshot from 2024-02-27 06-15-33](https://github.com/coddingtonbear/obsidian-web/assets/9726401/0212d49f-4e75-4c66-9ef4-3fabc4fb5571)

![Screenshot from 2024-02-27 06-26-30](https://github.com/coddingtonbear/obsidian-web/assets/9726401/2199bd75-8433-459e-8df8-329aeef77a4c)

![Screenshot from 2024-02-27 06-20-23](https://github.com/coddingtonbear/obsidian-web/assets/9726401/221d43b8-05bb-41e9-a30d-810152b07b61)

[obsidian-local-rest-api.bug_report.json](https://github.com/coddingtonbear/obsidian-web/files/14415617/obsidian-local-rest-api.bug_report.json)

![Screenshot from 2024-02-27 07-02-20](https://github.com/coddingtonbear/obsidian-web/assets/9726401/260b4a55-ba94-43c9-85a6-63309e776bd7)

### To Reproduce

2. Click on '...'
3. Scroll down to '...'

### Expected Behavior

Works sometimes. 

![image](https://github.com/coddingtonbear/obsidian-web/assets/9726401/13daaacd-5932-4d9c-9ba6-0e5d6a32a03d)

### Operating System

Linux Mint 21.3 Cinnamon

### Browser

chrome

### Browser Version

122.0.6261.69

**Comment by coddingtonbear** [comment]
You've given a lot of data up there -- thanks for that!  But, it's a little difficult for me to tell exactly _what_ you're finding difficulty with from the context above.  Is it:

- The dialog doesn't appear when you click the button.
- The dialog appears, but when you press the send button, nothing happens.
- The dialog appears, and when you press the send button, the dialog disappears, but the content doesn't appear in your notes.

I kind of wonder if the thing you're commenting on is that you see an "ERR" message on the extension icon, but have to say that I'm sort of surprised you see that message at all since your configuration shows that background search is disabled, and I'm fairly sure without background searches being on, we don't even try to do anything that might cause the extension icon to be decorated (and when we fail to do that, display such an error).  And, even in such a case, everything will still _work_, it just won't be able to let you know when it has found a match (but you have that disabled; so that's not surprising!).

Could you elaborate a bit on what exactly you're doing and what part of it isn't working?

**Reply by coddingtonbear** [reply]
Actually, I bet it is that last bit that you're referring to.  As it turns out, there is a bug that will cause us to _always_ do background searches, even when they're disabled.  I've put together a patch that (see https://github.com/coddingtonbear/obsidian-web/commit/33aa462e37fbae3f30d1c1f873f05d3bab2acc4a) that'll go out with the next release.

That being said, if this is truly the error you're referring to, things should still _work_; we just might show an erroneous error message on the icon.

Could you confirm that this is actually the problem you were referring to?

**Reply by BjornFJohansson** [reply]
Thank you for your reply and sorry for not including that last bit. Clicking on obsidian web would give me "Could not connect..." see below: 

![image](https://github.com/coddingtonbear/obsidian-web/assets/9726401/1a08b5a8-54c9-4171-8df2-2a1df5941ffd)

Going to the "Local REST API for Obsidian" and inserting my key would work perfectly. 

What happened next surprised me, I left it alone, waiting for a response here, tried it and it worked again, without any intervention. Maybe there is a lag somewhere or some restart that I did not do?

![image](https://github.com/coddingtonbear/obsidian-web/assets/9726401/203031d1-54f5-4d11-92d2-d5a7cd95472c)

Now it is working as intended and I can capture notes to obsidian. I cant really say if this was a bug or why it took the time it took, but maybe this could help others experiencing the same problem.

Thanks for your help!

**Reply by coddingtonbear** [reply]
That's... interesting.  I'm surprised that there would ever be any substantial lag around that.  I'm curious -- if you had to guess, how many files were in your notes vault?

**Comment by ffdumont** [comment]
[Uploading obsidian-local-rest-api.bug_report.json…]()
Hi @coddingtonbear ! I'm facing the same issue on MacOS 14.2.1 and  Chrome 122.0.6261.94 : get "Could not connect to Obsidian" message while setup is correct (green shield). I also observed it working intermittently this morning, but most of time it's not. Thank you !

**Reply by ffdumont** [reply]
Hi @coddingtonbear - any news on this ?

**Reply by coddingtonbear** [reply]
I do not provide support for Mint; so i’m afraid this discussion is out of my hands.

**Reply by ffdumont** [reply]
I used the previous issue but problems happens on Mac OS.

**Reply by coddingtonbear** [reply]
Hrm, well, I don't test things directly on Mac OS, but I can maybe give you some things to check on if you upload a bug report export from the extension options page's upper-right corner.  It looks like you tried to upload one to your original comment, but it doesn't appear that one is there.

**Reply by ffdumont** [reply]
[obsidian-local-rest-api.bug_report.json](https://github.com/coddingtonbear/obsidian-web/files/14664306/obsidian-local-rest-api.bug_report.json)
Here it is @coddingtonbear

**Reply by coddingtonbear** [reply]
Just at a glance, I can tell you that your version of Obsidian Local REST API is many, many versions out-of-date; I'm surprised you haven't seen the yellow banner on the extension's options page instructing you to update it.  More than likely, your certificate needs to be re-generated and re-imported, and once you do that update, you'll be instructed to do so in Obsidian Local REST API's settings.  From there, what I'd recommend is following these troubleshooting instructions I recently gave somebody: https://github.com/coddingtonbear/obsidian-web/discussions/174#discussioncomment-8740665

**Reply by ffdumont** [reply]
Thank you ! I did not get this update banner, but I followed you procedure and everything's fine again.

**Reply by coddingtonbear** [reply]
I'm fairly sure just updating the extension won't be sufficient, but if that works for now, good for you!  I'd imagine it'll start failing again once you've closed the options page for a while.  If I remember correctly, the root problem is the way earlier versions of Obsidian Local REST API were generating the certificate would make a certificate that would be accepted only in narrow situations (maybe only when the options page is open or was recently?).  I assume you'll have to follow the subsequent steps in the comment I linked to you to get things back to a clean slate.

**Comment by BjornFJohansson** [comment]
Still working after a few days...
